### PR TITLE
validate when gateway enabled and jaegerQuery is disabled

### PR DESCRIPTION
**Pull request by rubenvp8510**
@frzifus Is what we talked other day.

It make sense?

### CODE REVIEW

**Review by iblancasa — APPROVED** [review]

File: apis/tempo/v1alpha1/tempostack_webhook.go [hunk 1]
```diff
@@ -260,6 +260,17 @@ func (v *validator) validateQueryFrontend(tempo TempoStack) field.ErrorList {
 	return nil
 }
 
+func (v *validator) validateGateway(tempo TempoStack) field.ErrorList {
+	if tempo.Spec.Components.Gateway.Enabled && !tempo.Spec.Components.QueryFrontend.JaegerQuery.Enabled {
+		path := field.NewPath("spec").Child("components").Child("gateway").Child("enabled")
+		return field.ErrorList{
+			field.Invalid(path, tempo.Spec.Components.Gateway.Enabled,
+				"gateway require enable jaeger query to work.",
```
Comment (andreasgerstmayr) [inline apis/tempo/v1alpha1/tempostack_webhook.go:268]: I would reword the error message a bit, for example
```suggestion
				"to use the gateway, please enable jaegerQuery",
```
or
```suggestion
				"gateway cannot be enabled if jaegerQuery is disabled",
```

**Review by andreasgerstmayr — APPROVED** [review]
